app/records/models.py

```diff
@@ -344,33 +349,49 @@ def unpublished_finding_aids(self) -> list[str]:
     @cached_property
     def hierarchy(self) -> tuple[Record, ...]:
         """Returns tuple of records transformed from the values of the attr if found, empty tuple otherwise."""
-        return tuple(
-            Record(item)
-            for item in self.get("@hierarchy", ())
-            if item.get("identifier")
-        )
+        hierarchy_records = ()
+        for hierarchy_item in self.get("@hierarchy", ()):
+            if hierarchy_item.get("identifier"):
+                # page_record_is_tna: carry status to hierarchy record
+                hierarchy_records += (
+                    Record(
+                        hierarchy_item | {"page_record_is_tna": self.is_tna}
```

Comment:
Should this be done by us or do we need to raise this as an issue with the API?

Reply:
Valid question. Will check and we can discuss when they are back from leave. 

It was a dilemma. I thought since its just one field over keeping size of response low and some learnings from OHOS. The `groupArray` attribute would be repeated for each hierarchy including next, previous - this would increase the api response size. Also there was another bits from OHOS where we could not get a single return value across buckets so  `groupArray` was given https://github.com/nationalarchives/ds-ohos-wagtail/blob/932a401cc3cd8e47ef223ebb662c29b5a65f41de/etna/records/models.py#L640-L653

Reply:
Check with them but this definitely feels like an enhancement to the API. Adding just a `page_record_is_tna` to each `Record` object in the response (`hierarchy`, `previous`, `next`) shouldn't increase the size too much.